ikuzo/domain/organization.go

```diff
@@ -109,3 +124,37 @@ func (id OrganizationID) Valid() error {
 
 	return nil
 }
+
+// String returns the OrganizationID as a string
+func (id OrganizationID) String() string {
+	return string(id)
+}
+
+// RawID returns the raw direct identifier string for an Organization
+func (o *Organization) RawID() string {
+	return o.ID.String()
+}
+
+func (o *Organization) NewDatasetURI(spec string) string {
+	return fmt.Sprintf(o.Config.MintDatasetURL, o.Config.RDFBaseURL, spec)
+}
+
+// GetOrganizationID retrieves an OrganizationID from a *http.Request.
+//
+// This orgID is set by middleware and available for each request
+func GetOrganizationID(r *http.Request) OrganizationID {
+	orgID := r.Context().Value(orgIDKey{})
+	if orgID != nil {
+		id, _ := NewOrganizationID(orgID.(string))
+		return id
+	}
+
+	return ""
+}
+
+// SetOrganizationID sets the orgID in the context of a *http.Request
+//
+// This function is called by the middleware
+func SetOrganizationID(r *http.Request, orgID string) *http.Request {
+	return r.WithContext(context.WithValue(r.Context(), orgIDKey{}, orgID))
```

Comment:
De WithContext maakt een shallow copy van de request. Is het niet verstandiger hier een r.Clone(...) te doen?